src/integrationTests/options.test.ts

```diff
@@ -209,11 +219,332 @@ describe('useCookie option', () => {
+        useCookie: true,
+      });
+    });
+    test('should use new base url', () => {
+      uid2.init({
+        baseUrl: newBaseUrl,
+      });
+      const configCookie = getConfigCookie();
+      expect(configCookie).toHaveProperty('baseUrl', newBaseUrl);
+    });
+
+    test('should use old base url', () => {
+      uid2.init({
+        cookiePath: cookiePath,
+      });
+      const configCookie = getConfigCookie();
+      expect(configCookie).toHaveProperty('baseUrl', oldBaseUrl);
+    });
+  });
+
+  describe('new identity provided but expired', () => {
+    const newIdentity = makeIdentity({ refresh_expires: Date.now() - 100000 });
+    const useCookie = true;
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        useCookie: useCookie,
+      });
+      uid2.init({
+        identity: newIdentity,
+      });
+    });
+    test('should set value to old identity', () => {
+      expect(getUid2(useCookie).advertising_token).toBe(identity.advertising_token);
+    });
+    test('should set refresh timer once', () => {
+      expect(setTimeout).toHaveBeenCalledTimes(1);
+      expect(clearTimeout).not.toHaveBeenCalled();
+    });
+    test('old ideneity should be in available state', () => {
+      (expect(uid2) as any).toBeInAvailableState(identity.advertising_token);
+    });
+  });
+
+  describe('new identity provided but expires before old identity', () => {
+    const oldIdentity = makeIdentity({ refresh_expires: Date.now() + 5000 });
+    const newIdentity = makeIdentity({ refresh_expires: Date.now() + 100000 });
+    const useCookie = true;
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: oldIdentity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        useCookie: useCookie,
+      });
+      uid2.init({
+        identity: newIdentity,
+      });
+    });
+
+    test('should set value', () => {
+      expect(getUid2(useCookie).advertising_token).toBe(oldIdentity.advertising_token);
+    });
+    test('should set refresh timer', () => {
+      expect(setTimeout).toHaveBeenCalledTimes(1);
+      expect(clearTimeout).not.toHaveBeenCalled();
+    });
+    test('should be in available state', () => {
+      (expect(uid2) as any).toBeInAvailableState(oldIdentity.advertising_token);
+    });
+  });
+
+  describe('new identity provided and expires after old identity', () => {
+    const newIdentity = makeIdentity();
+    const useCookie = true;
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        useCookie: useCookie,
+      });
+      uid2.init({
+        identity: newIdentity,
+      });
+    });
+
+    test('should set value', () => {
+      expect(getUid2(useCookie).advertising_token).toBe(newIdentity.advertising_token);
```

Comment:
Again, the two identities have the same advertising_token so you can't actually tell which one is being used. I can understand the mistake - I think it'd be a good idea to change makeIdentity to add a random number to the end of both tokens so they're always different.

Reply:
My mistake - I thought every time `makeIdentity` was called, that it generated a random advertising token.  I've changed the `advertising_token` variable in all the new identities throughout the tests